tiledb/sm/filesystem/ls_callback.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file ls_callback.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This defines the type definition used for ls callback functions in VFS.
+ */
+
+#ifndef TILEDB_LS_CALLBACK_H
+#define TILEDB_LS_CALLBACK_H
+
+#include <cstdint>
+#include <functional>
+#include <stdexcept>
+
+template <class F>
+concept LsCb = true;
+
+namespace tiledb::sm {
+
+using LsCallback =
+    std::function<bool(const std::string_view&, uint64_t, void*)>;
```

Comment:
Is this dead code?

Reply:
Yes it is, I meant to call this out in my comments but may not have. I ended up using this signature for the `LsCallbackWrapper::operator()`, but didn't use this directly. I was expecting more changes so I left it around in case we found a use for it that I wasn't seeing, but I can remove if it's still not needed.